pages/api/user/index.ts

```diff
@@ -0,0 +1,178 @@
+import { PrismaClient } from '@prisma/client';
+import type { NextApiRequest, NextApiResponse } from 'next';
+import { ApiResponse, EmptyPayload, StatusMessageType } from '../../../types/apiResponse';
+import { AccountPostData } from '../../../types/user';
+import { Nullable } from '../../../types/utils';
+import { getUserFromJwt, UserDetailsFromRequest, withAuthUser } from '../../../utils/auth/jwtHelpers';
+import { createJsonResponse, HttpMethod, HttpStatus, rejectHttpMethod } from '../../../utils/http/httpHelpers';
+import { withPrismaErrorHandling } from '../../../utils/prisma/prismaHelpers';
+
+const prisma = new PrismaClient();
+
+/** Limited set of data representing user and the list of user application IDs. Only for use in this endpoint. */
+interface UserData {
+  uid: string;
+}
+
+enum MessageType {
+  // POST link accounts.
+  NEW_USER_LINKED,
+  NEW_USER_UNVERIFIED,
+  OLD_USER_NOT_FOUND,
+  NEW_USER_EXISTS,
+  INVALID_OLD_TOKEN,
+  OLD_USER_VERIFIED,
+  // POST account creation only.
+  USER_CREATED,
+  USER_ALREADY_EXISTS,
+  // DELETE accounts
+  USER_DELETED,
+  DELETE_TARGET_NOT_FOUND,
+}
+
+const messages = Object.freeze({
```

Comment:
I made a lot of changes to the messages to make it user facing. Feel free to veto and resolve if you want.

Reply:
Thanks!